src/components/MobileNav/NavList.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from "react";
+import { Link } from "react-router-dom";
+import ReactSVG from "react-svg";
+
+import { baseUrl } from "../App/routes";
+import NavItem from "./NavItem";
+
+interface NavListProps {
+  items: NavItem[];
+  hideOverlay(): void;
+}
+
+interface NavListState {
+  parent: NavItem | null;
+  displayedItems: NavItem[];
+}
+
+class NavList extends React.PureComponent<NavListProps, NavListState> {
+  constructor(props: NavListProps) {
```

Comment:
I think that the convention here is not using the constructor, but setting state as class property
```
state: NavListState = {
  displayedItems: this.props.items,
  parent: null
};
```